packages/flutter/lib/src/material/page.dart

```diff
@@ -115,23 +115,26 @@ class _CupertinoBackGestureController extends NavigationGestureController {
   }) : super(navigator);
 
   AnimationController controller;
-  VoidCallback onDisposed;
+  final VoidCallback onDisposed;
 
   @override
   void dispose() {
-    super.dispose();
-    onDisposed();
     controller.removeStatusListener(handleStatusChanged);
     controller = null;
+    onDisposed();
+    super.dispose();
   }
 
   @override
   void dragUpdate(double delta) {
-    controller.value -= delta;
+    controller?.value -= delta;
```

Comment:
The Scaffold is in charge of driving the navigation gesture, so it starts the gesture and feeds events to it, but it doesn't own the navigation gesture object.  Normally, this isn't a problem because the Scaffold is a part of the route and so it dies when the route dies, but if you keep the Scaffold alive (e.g., using global key reparenting), then you might try to keep feeding events to a navigation gesture for a route that was dead.  This will now trigger asserts.

I think the proper solution is to have the scaffold own the navigation gesture object and to be responsible for aborting the navigation gesture if it gets reparented into a different route.  I started down that path, and I think it will work, but I pushed the eject button before I finished.